crypto/ct/ct_sct.c

```diff
@@ -317,12 +317,11 @@ int SCT_set_source(SCT *sct, sct_source_t source)
 
 int SCT_LIST_set_source(const STACK_OF(SCT) *scts, sct_source_t source)
 {
-    int i, ret = 1;
+    int i, ret = 0;
 
     for (i = 0; i < sk_SCT_num(scts); ++i) {
-        ret = SCT_set_source(sk_SCT_value(scts, i), source);
-        if (ret != 1)
-            break;
+        if (SCT_set_source(sk_SCT_value(scts, i), source))
```

Comment:
Next question. I' m wondering about `SCT_set0_log`. What's the object lifetime story here? This suggests that the logs in the CTLOG_STORE must outlive the SCTs, which isn't documented anywhere, and also seems dangerous.

A non-const `CTLOG *CTLOG_STORE_get0_log_by_id` also seems dubious.